GoogleForCreators/web-stories-wp
# Dashboard: Add story context menu to list view

## Summary
Until now we've only had the 'context menu' (three dots that show up on hover of a story in the grid) on the grid view. This PR adds that menu on hover of story title in list view with same functionality as grid. 

## Relevant Technical Choices
Before this, it made sense that the context menu lived inside the storyGrid but now not so much - 

- Pulled the `contextMenu` from the `cardGridItem` into its own directory called `storyMenu` since this menu isn't so much tied to the grid item as it is a story. 
- Pulled the inline input that is swapped with the cardTitle into its own directory called 'inlineInputForm' and added a hidden label to it. 
- Moved the logic for setting up the context menu up to views/myStories/content/storiesView  - this allowed a lot of clean up in the shared view components. It also means I had to touch a few extra files to catch up various storybook components and saved templates.

## User-facing changes
- Adds context menu (storyMenu) to list view of my stories. So now when you hover or focus on a story title in the list view you see the menu button. When you expand it you have same functionality (it's the same exact menu) as the grid. 
- I did change the alignment of the context menu button on the grid, thoughts on having the button to open the storyMenu align with the top of the story title? I feel like this is cleaner. 
![Screen Shot 2020-06-01 at 4 34 15 PM](https://user-images.githubusercontent.com/10720454/83464530-f9ce1800-a425-11ea-8ccb-79d26ffc03a4.png)



## Testing Instructions
- Context menu works in grid and list view still
- Note that the following actions are not connected yet: `preview` & `create template` 
- But you should be able to `open in editor`, `rename`, `duplicate`, `delete`

**Copy functionality**
![copy in list view](https://user-images.githubusercontent.com/10720454/83450736-b0240400-a40a-11ea-82ff-f281858c0a90.gif)

**Rename functionality**
![list view rename](https://user-images.githubusercontent.com/10720454/83450718-a7333280-a40a-11ea-93d5-6417d3d3a0fd.gif)

**Delete functionality**
![delete in list view](https://user-images.githubusercontent.com/10720454/83450730-ac907d00-a40a-11ea-9e20-47f740bb07b5.gif)


---



Fixes #2124
